server_test.go

```diff
@@ -124,3 +126,104 @@ func TestServerClose(t *testing.T) {
 		return
 	}
 }
+
+func TestServerClose_ConnectionLeak(t *testing.T) {
```

Comment:
This test is a bit finicky, but without the fix it will time out (as a proof of the issue being fixed).